agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1167,24 +1168,52 @@ func (dg *dockerGoClient) Stats(id string, ctx context.Context) (<-chan *docker.
 		return nil, err
 	}
 
-	stats := make(chan *docker.Stats)
-	options := docker.StatsOptions{
-		ID:                id,
-		Stats:             stats,
-		Stream:            true,
-		Context:           ctx,
-		InactivityTimeout: StatsInactivityTimeout,
-	}
+	returnStats := make(chan *docker.Stats)
 
-	go func() {
-		statsErr := client.Stats(options)
-		if statsErr != nil {
-			seelog.Infof("DockerGoClient: Unable to retrieve stats for container %s: %v",
-				id, statsErr)
+	if !dg.config.PollMetrics {
+		seelog.Infof("DockerGoClient: Starting to Stream for metrics")
+		options := docker.StatsOptions{
+			ID:                id,
+			Stats:             returnStats,
+			Stream:            true,
+			Context:           ctx,
+			InactivityTimeout: StatsInactivityTimeout,
 		}
-	}()
+	
+		go func() {
+			statsErr := client.Stats(options)
+			if statsErr != nil {
+				seelog.Infof("DockerGoClient: Unable to retrieve stats for container %s: %v",
+					id, statsErr)
+			}
+		}()
+	} else {
+		seelog.Infof("DockerGoClient: Starting to Poll for metrics")
+		//Sleep for a random period time up to the polling interval. This will help make containers ask for stats at different times
+		time.Sleep(time.Second * time.Duration(rand.Intn(int(dg.config.PollingMetricsWaitDuration.Seconds()))))
```

Comment:
Does this mean this could end up sleeping for the whole wait duration, skipping a cycle of poll metrics? not sure if this could lead to any unexpected behavior